src/test/java/com/github/javafaker/InternetTest.java

```diff
@@ -105,4 +106,109 @@ public void testPasswordMinLengthMaxLengthIncludeUpperCaseIncludeSpecial() {
+    public void testMacAddress() {
+        assertThat(faker.internet().macAddress(), countOf(':', is(5)));
+        assertThat(faker.internet().macAddress(""), countOf(':', is(5)));
+
+        assertThat(faker.internet().macAddress("fa:fa:fa"), startsWith("fa:fa:fa"));
+        assertThat(faker.internet().macAddress("fa:fa:fa"), countOf(':', is(5)));
+
+        assertThat(faker.internet().macAddress("01:02"), startsWith("01:02"));
+        assertThat(faker.internet().macAddress("01:02"), countOf(':', is(5)));
+
+        // loop through 1000 times just to 'run it through the wringer'
+        for (int i=0; i<1000;i++) {
+            assertThat(
+              "Is valid mac format",
+              faker.internet().macAddress(),
+              matchesRegularExpression("[0-9a-fA-F]{2}(\\:([0-9a-fA-F]{1,4})){5}"));
+        }
+    }
+
+
+    @Test
+    public void testIpV4Address() {
+        assertThat(faker.internet().ipV4Address(), countOf('.', is(3)));
+        for (int i = 0; i < 100; i++) {
+            final String[] octets = faker.internet().ipV4Address().split("\\.");
+            assertThat("first octet is 1-255", parseInt(octets[0]),
+                    both(greaterThan(0)).and(lessThanOrEqualTo(255)));
+            assertThat("second octet is 0-255", parseInt(octets[1]),
+                    both(greaterThanOrEqualTo(0)).and(lessThanOrEqualTo(255)));
+            assertThat("second octet is 0-255", parseInt(octets[2]),
+                    both(greaterThanOrEqualTo(0)).and(lessThanOrEqualTo(255)));
+            assertThat("second octet is 0-255", parseInt(octets[3]),
+                    both(greaterThanOrEqualTo(0)).and(lessThanOrEqualTo(255)));
+        }
+    }
+
+    @Test
+    public void testIpV4Cidr() {
+        assertThat(faker.internet().ipV4Cidr(), countOf('.', is(3)));
+        assertThat(faker.internet().ipV4Cidr(), countOf('/', is(1)));
+        
+        for (int i = 0; i < 1000; i++) {
+            assertThat(parseInt(faker.internet().ipV4Cidr().split("\\/")[1]),
+                    both(greaterThanOrEqualTo(1)).and(lessThan(32)));
+        }
+    }
+
+    @Test
+    public void testPrivateIpV4Address() {
+        String ten_dot = "^10\\..+";
+        String one_two_seven = "^127\\..+";
+        String one_six_nine = "^169\\.254\\..+";
+        String one_nine_two = "^192\\.168\\..+";
+        String one_seven_two = "^172\\.(16|17|18|19|20|21|22|23|24|25|26|27|28|29|30|31)\\..+";
+        for (int i = 0; i < 1000; i++) {
+            String addr = faker.internet().privateIpV4Address();
+            
+            assertThat(addr, anyOf(matchesRegularExpression(ten_dot),
+                    matchesRegularExpression(one_two_seven),
+                    matchesRegularExpression(one_six_nine),
+                    matchesRegularExpression(one_nine_two),
+                    matchesRegularExpression(one_seven_two)));
+        }
+    }
+
+    @Test
+    public void testPublicIpV4Address() {
+        String ten_dot = "^10\\.";
```

Comment:
yeah. let me get to that. sorry, i copied the variables from the ruby tests and forgot to update that.